pmd-java/src/test/resources/net/sourceforge/pmd/lang/java/rule/performance/xml/AddEmptyString.xml

```diff
@@ -50,6 +50,72 @@ public class Foo {
     @Deprecated(since = "" + VERSION)
     public void foo() {
     }
+}
+        ]]></code>
+    </test-code>
+
+    <test-code>
+        <description>[java] AddEmptyString - false negative with empty var #3625</description>
+        <expected-problems>1</expected-problems>
+        <expected-linenumbers>4</expected-linenumbers>
+        <code><![CDATA[
+public class Main {
+    public void func() {
+        final String var = "";
+        String test = var + "123"; // should report a warning here
+    }
+}
+        ]]></code>
+    </test-code>
+
+    <test-code>
+        <description>Local Variables be empty String</description>
+        <expected-problems>3</expected-problems>
+        <expected-linenumbers>8,9,10</expected-linenumbers>
+        <code><![CDATA[
+class Main {
+    public static void main(String[] args) {
+        final String string1 = "";
+        final String string2 = "";
+        final String string3 = "";
+        String string4 = "";
+
+        String a = string1 + 114;
+        String b = string2 + 514;
+        String c = string3 + 1919810;
+
+        string4 = "foo";
+        String d = string4 + 1; // should not be flagged, because string4 is not empty anymore
```

Comment:
actually we only consider final vars, that are initialized with an empty string... so we don't look for other assignments